db/db_impl/db_impl.cc

```diff
@@ -562,6 +562,19 @@ Status DBImpl::CloseHelper() {
     delete txn_entry.second;
   }
 
+  std::unordered_set<std::string> db_paths;
+  db_paths.insert(dbname_);
+  for (auto& db_path : initial_db_options_.db_paths) {
+    db_paths.insert(db_path.path);
+  }
+  auto cfs = versions_->GetColumnFamilySet();
+  for (auto it = cfs->begin(); it != cfs->end(); ++it) {
+    for (auto& cf_path : (*it)->ioptions()->cf_paths) {
```

Comment:
For those auto in this patch, my vote is to spell out the type. Google C++ Style has a rule of auto which allows auto if it make the code clearer to readers: https://google.github.io/styleguide/cppguide.html#Type_deduction

I personally feels those auto makes it harder for me to read, rather than easier.